streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -215,6 +212,10 @@ Processor describe() {
         }
     }
 
+    // Map from topics to their matched regex patterns, this is to ensure one topic is passed through on source node
+    // even if it can be matched by multiple regex patterns. Only used by SourceNodeFactory
+    private final Map<String, Pattern> topicToPatterns = new HashMap<>();
```

Comment:
I think in a previous commit this was `static` and I had concerns that it wouldn't work, so I'm glad to see you changed this.